docs/docs/introduction/getting-started.mdx

```diff
@@ -37,91 +37,121 @@ bower install --save recoil
 ## RecoilRoot
 
 Components that use recoil state need `RecoilRoot` to appear somewhere in the parent tree. A good place to put this is in your root component:
+This can also be your index.js file.
+index.js
 
 ```jsx
-import React from 'react';
-import {
-  RecoilRoot,
-  atom,
-  selector,
-  useRecoilState,
-  useRecoilValue,
-} from 'recoil';
-
-function App() {
-  return (
-    <RecoilRoot>
-      <CharacterCounter />
-    </RecoilRoot>
-  );
-}
+    import React from 'react';
+    import ReactDOM from 'react-dom/client';
+    import './index.css';
+    import App from './App';
+    import {RecoilRoot} from 'recoil'
+
+
+    const root = ReactDOM.createRoot(document.getElementById('root'));
+    root.render(
+      <React.StrictMode>
+          <RecoilRoot>
+              <App />
+          </RecoilRoot>
+      </React.StrictMode>
 ```
 
-We'll implement the `CharacterCounter` component in the following section.
+You will need to have an App.js file component that you imported to the index.js file.
+The App.js file will take in the following component TexInput, CharCout
 
-## Atom
-
-An **atom** represents a piece of **state**. Atoms can be read from and written to from any component. Components that read the value of an atom are implicitly **subscribed** to that atom, so any atom updates will result in a re-render of all components subscribed to that atom:
+import './App.css';
+import TextInput from './TextInput';
+import CharCout from './CharCout';
 
-```javascript
-const textState = atom({
-  key: 'textState', // unique ID (with respect to other atoms/selectors)
-  default: '', // default value (aka initial value)
-});
-```
 
-Components that need to read from _and_ write to an atom should use `useRecoilState()` as shown below:
+const App = () => {
 
-```jsx
-function CharacterCounter() {
   return (
-    <div>
-      <TextInput />
-      <CharacterCount />
-    </div>
+         <main>
+              <TextInput /> 
+              <CharCout />
+         </main>
+   
   );
 }
 
-function TextInput() {
-  const [text, setText] = useRecoilState(textState);
+export default App;
 
-  const onChange = (event) => {
-    setText(event.target.value);
-  };
+We will then naviagate to our src folder to create Those components
+We'll implement the `CharacterCounter` component in the following section.
 
-  return (
-    <div>
-      <input type="text" value={text} onChange={onChange} />
-      <br />
-      Echo: {text}
-    </div>
-  );
-}
+## Atom
+
+An **atom** represents a piece of **state**. Atoms can be read from and written to from any component. Components that read the value of an atom are implicitly **subscribed** to that atom, so any atom updates will result in a re-render of all components subscribed to that atom:
+
+We would start here with the TextInput component
+
+```javascript
+    import React from 'react'
+    import {atom, useRecoilState} from 'recoil';
+
+    export const inputState = atom({
```

Comment:
Part of the intention of the existing documentation breaking up the `inputState` and `TextInput` definitions is that they could be in separate files.  It can be a good practice to introduce separating the state definitions and React components into separate files if they are used as shared state.